spec.html

```diff
@@ -37159,28 +37161,91 @@
+                                  1. Let _genv_ be the running execution context’s VariableEnvironment.
+                                  2. Let _benv_ be the running execution context’s LexicalEnvironment.
+                                  3. Let _fobj_ be _benv_.GetBindingValue(_F_, *false*).
+                                  4. Assert: _fobj_ is never an abrupt completion.
+                                  5. Let _status_ be _genv_.SetMutableBinding(_F_, _fobj_, *false*).
+                                  6. ReturnIfAbrupt(_status_).
+            </emu-alg>
+          </emu-annex>
+          <emu-annex id="sec-web-compat-evaldeclarationinstantiation">
+            <h1>Changes to EvalDeclarationInstantiation</h1>
+            <p>During EvalDeclarationInstantiation (<emu-xref href="#sec-evaldeclarationinstantiation"></emu-xref>) the following steps are performed in place of step 12:</p>
+            <emu-alg>
+              1. If _strict_ is *false*, then
+                  1. For each |FunctionDeclaration| _f_ that is directly contained in the |StatementList| of a |Block|, |CaseClause|, or |DefaultClause| Contained within _body_,
+                      1. Let _F_ be StringValue of the |BindingIdentifier| of |FunctionDeclaration| _f_.
+                      2. If replacing the |FunctionDeclaration| _f_ with a |VariableStatement| that has _F_ as a |BindingIdentifier| would not produce any Early Errors for _body_, then
+                          1. Let _bindingExists_ be *false*.
+                          2. Let _fnDefinable_ be *true*.
+                          3. Let _thisLex_ be _lexEnv_.
+                          4. Assert: the following loop will terminate.
+                          5. Repeat while _thisLex_ is not the same as _varEnv_,
+                              1. Let _thisEnvRec_ be _thisLex_'s EnvironmentRecord.
+                              2. If _thisEnvRec_ is not an object Environment Record, then
+                                1. If _thisEnvRec_.HasBinding(_F_) is *true*, then
+                                  1. Let _bindingExists_ be *true*.
+                              1. Let _thisLex_ be _thisLex_'s outer environment reference.
+                          6. If _bindingExists_ is *true* and _varEnvRec_ is a global Environment Record, then
+                              1. If _envRec_.HasLexicalDeclaration(_F_) is *false*, then
+                                  1. Let _fnDefinable_ be _envRec_.CanDeclareGlobalFunction(_F_).
+                                  1. ReturnIfAbrupt(_fnDefinable_).
+                          7. If _bindingExists_ is *false* and _fnDefinable_ is *true* , then
+                              1. If _declaredVarNames_ does not contain _F_, then
+                                  1. Append _F_ to _declaredVarNames_.
```

Comment:
Yeah I am in the process of doing that (although I'm not going to reuse the declaredVarNames machinery at all - if _envRec_ is not a global Environment Record I'll do a create/setMutableBinding.